### ISSUE TITLE
Release AUR package

### ISSUE THREAD

**Issue by aaronleopold** [aaronleopold/stump#98]
### Discussed in https://github.com/aaronleopold/stump/discussions/97

<div type='discussions-op-text'>

<sup>Originally posted by **Relwi** March 15, 2023</sup>
Hey, is there the possibility that we have in a future an AUR package? </div>

At some point, it would be a nice to have if CI can also publish the zipped(?) linux binary + webapp bundle to AUR.

**Comment by airvzxf** [comment]
Hi, I am interested in this enhancement, but I am confident that it needs more details to achieve the goal.

My first question is: What should be the zipped Linux binary + web app bundle?
- In the [GitHub actions](https://github.com/stumpapp/stump/tree/main/.github/actions), I noticed that you create different builds. I assume that the [Desktop build](https://github.com/stumpapp/stump/blob/main/.github/actions/build-desktop/action.yml) should be gone to the AUR. Is it correct?
- What are you expected to add to AUR as a web app bundle?

My second question is: How do you handle the versions for all the builds?
- I noticed that the version is 0.0.0 for all the builds. In Rust, the version is in the [stump/Cargo.toml](https://github.com/stumpapp/stump/blob/5d3b1f67699950d65a656bbd6cde7e255ed6f2ff/Cargo.toml#L12).
- I am not seen GitHub releases or tags. In docker images, I saw only the [nightly version and the experimental](https://hub.docker.com/r/aaronleopold/stump).
- How do you want to handle the versions in AUR? Stump-nightly and stump-experimental. Or other?

---

These are my AUR:
- [optional-packages](https://aur.archlinux.org/packages/optional-packages).
- [battery-discharging](https://aur.archlinux.org/packages/battery-discharging).
- [brightness-xrandr](https://aur.archlinux.org/packages/brightness-xrandr).

**Comment by aaronleopold** [comment]
> My first question is: What should be the zipped Linux binary + web app bundle?

Exactly correct, the binary and the webapp bundle zipped together. Not sure if an installer of some kind would be necessary, but those two items are all that is otherwise needed.

> In the [GitHub actions](https://github.com/stumpapp/stump/tree/main/.github/actions), I noticed that you create different builds. I assume that the [Desktop build](https://github.com/stumpapp/stump/blob/main/.github/actions/build-desktop/action.yml) should be gone to the AUR. Is it correct?

Eventually, yes it can. Until the first more stable release of Stump is released, though, it shouldn't. I'm not personally focusing on ensuring the desktop app is always functional right now, since there are many other higher priority things I need to get out the door first.

> What are you expected to add to AUR as a web app bundle?

That's a big question, and one I probably can't answer. Someone else requested AUR releases.

> My second question is: How do you handle the versions for all the builds?

I don't. This is something that I have an issue for actually: https://github.com/stumpapp/stump/issues/93

Basically, the release workflow is not yet implemented. Ideally, I would imagine when tags are pushed from main to:

1. Start various builds
2. Generate a change log

In general, I don't think any _actual_ publishing to AUR should be done yet. You're welcome to continue with this ticket as far or as little as you'd like, but keep in mind it should just be adding the required CI/hookups for it while keeping it 'disabled' so nothing gets published until versioning is a thing here

**Comment by airvzxf** [comment]
Thanks for the detailed answer.

I have two questions before continue with this ticket (PR and RM).
1. If I create the PR, will you merge it if it complains your basic expectations? I am interested in the Hacktoberfest challenge.
2. I can solve it, in the similar way that I solved in my AUR packages, which for me works good. Do you agree with this solution?
  - Basically, I automate the creation of the build as you did, then upload the information to the AUR. Because, AUR doesn't store files, only links or commands, and then it downloads and executes these commands in Arch Linux.
  - Please, review the following links:
    - [.github/actions/aur-deployment/](https://github.com/airvzxf/archlinux-optional-packages/tree/main/.github/actions/aur-deployment)
    - [.github/workflows/deploy-to-aur.yml](https://github.com/airvzxf/archlinux-optional-packages/blob/main/.github/workflows/deploy-to-aur.yml)

**Comment by aaronleopold** [comment]
> If I create the PR, will you merge it if it complains your basic expectations? I am interested in the Hacktoberfest challenge.

Yep! To reiterate, the general expectation would be to add the integration that is functional (i.e. it can run a build and push info to the AUR) but have the workflow be disabled until I am ready (i.e. versioned releases start). That way, there is less of a hurdle once this functionality is needed and we don't pollute the AUR with nightly builds.

> I can solve it, in the similar way that I solved in my AUR packages, which for me works good. Do you agree with this solution?

I trust your judgement. Your description sounds logical to me:

> Basically, I automate the creation of the build as you did, then upload the information to the AUR 

Keeping in mind that while I have primarily used arch-based distributions, my knowledge about best practices for AUR packaging is lacking 😅

**Comment by airvzxf** [comment]
Ok, would you kindly assign to me? I'll start working in a few minutes or hours.

**Comment by airvzxf** [comment]
I am trying to follow the [Development Guide](https://github.com/stumpapp/stump#developer-guide-), but it is not working. Do you have an idea how to solve it?

Error:
```
WARN Unsupported engine: wanted: {"node":"18.16.0"} (current: {"node":"v20.8.0","pnpm":"8.8.0"})

packages/interface | WARN The field "pnpm" was found in /home/wolf/workspace/projects/stump/packages/interface/package.json. This will not take effect. You should configure "pnpm" at the root of the workspace instead.

Scope: all 12 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +2209

Running install script...
 ELIFECYCLE  Command failed.
 ELIFECYCLE  Command failed with exit code 1.
```

I attached the screenshots.

![Screenshot_2023-10-04_07-42-18](https://github.com/stumpapp/stump/assets/831380/94c4cc89-c89e-4300-88d1-097ecb42fd97)

---

![Screenshot_2023-10-04_07-42-39](https://github.com/stumpapp/stump/assets/831380/e83cf5e4-3946-4fa1-9545-d89e123acd2f)

---

![Screenshot_2023-10-04_07-43-11](https://github.com/stumpapp/stump/assets/831380/7aa66e94-204b-4c74-bede-326670a3d727)

**Comment by aaronleopold** [comment]
Are there any log files being generated? The error code alone isn't much to go off of. You can maybe try removing all `node_modules` and then:

```bash
pnpm store prune
pnpm i --ignore-scripts
```

I'd also be curious if trying to use `moon` would present the same issue for you

**Comment by airvzxf** [comment]
> I'd also be curious if trying to use `moon` would present the same issue for you. Could also try removing the `pnpm-lock.yaml` and retrying the above.

First, I deleted these files, then tried `pnpm i` and I got the same error.

```bash
$ find . -iname pnpm-lock.yaml
./node_modules/.pnpm/remark-reading-time@2.0.1/node_modules/remark-reading-time/pnpm-lock.yaml
./pnpm-lock.yaml

$ rm -f ./node_modules/.pnpm/remark-reading-time@2.0.1/node_modules/remark-reading-time/pnpm-lock.yaml
$ rm -f ./pnpm-lock.yaml

$ pnpm i
```

---

> Are there any log files being generated? The error code alone isn't much to go off of. You can maybe try removing all `node_modules` and then:
> 
> ```shell
> pnpm store prune
> pnpm i --ignore-scripts
> ```
 I restored the file `./pnpm-lock.yaml` then I tried these commands and it worked.

![Screenshot_2023-10-04_10-53-50](https://github.com/stumpapp/stump/assets/831380/9365d0d0-2b24-4e5f-912e-96aff7eef507)

---

> Are there any log files being generated? The error code alone isn't much to go off of. You can maybe try removing all `node_modules` and then:
> 
> ```shell
> pnpm store prune
> pnpm i --ignore-scripts
> ```

I tried first delete the file `pnpm-lock.yaml` then I executed these commands and it worked too.

![Screenshot_2023-10-04_11-06-30](https://github.com/stumpapp/stump/assets/831380/0d202894-b85a-4913-9e86-cde4998f67ad)

![Screenshot_2023-10-04_11-06-47](https://github.com/stumpapp/stump/assets/831380/b873d7a1-9522-4215-a229-69e21142bf8c)

![Screenshot_2023-10-04_11-07-07](https://github.com/stumpapp/stump/assets/831380/371038fc-b2fe-4e1f-814b-db955574d5f3)

**Comment by airvzxf** [comment]
New error, I ran `pnpm dev:desktop`.

These are the outputs. 

First time:

![Screenshot_2023-10-04_11-25-38](https://github.com/stumpapp/stump/assets/831380/3769a42c-f8c1-4aa2-a5bc-bf222f1caac5)

---

Second time:

![Screenshot_2023-10-04_11-27-11](https://github.com/stumpapp/stump/assets/831380/101924cb-eb2e-49c9-a76a-fe8a7e38696c)

**Comment by aaronleopold** [comment]
> `cargo-watch: command not found`

To run any of the dev scripts, you'll need to install that dependency (step 2 of the install guide). However, you don't need to worry about the desktop for now. If you remember:

> > In the [GitHub actions](https://github.com/stumpapp/stump/tree/main/.github/actions), I noticed that you create different builds. I assume that the [Desktop build](https://github.com/stumpapp/stump/blob/main/.github/actions/build-desktop/action.yml) should be gone to the AUR. Is it correct?

> Eventually, yes it can. Until the first more stable release of Stump is released, though, it shouldn't. I'm not personally focusing on ensuring the desktop app is always functional right now, since there are many other higher priority things I need to get out the door first.

I'm mostly concerned with the server and web app bundle for now, sorry if that wasn't communicated well 😅 but that shrinks the scope a bit for this ticket

**Comment by airvzxf** [comment]
Yes, I agree. I'm deviating from the goal. Furthermore, I was thinking this day that this ticket is not ready to start because at the end AUR packages have a dependency on digital storage in some place (GitHub releases, hosting, or other). Then, I suggest unassigning me and creating some kind of block.

**Comment by aaronleopold** [comment]
A community-maintained AUR package now exists: https://www.stumpapp.dev/docs/getting-started/installation/source#arch-linux-aur
